csaf-validation/src/main/kotlin/io/github/csaf/sbom/validation/tests/Tests.kt

```diff
@@ -741,6 +782,75 @@ object Test6123MultipleUseOfSameCVE : Test {
     }
 }
 
+/**
+ * Implementation of
+ * [Test 6.1.24](https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#6124-multiple-definition-in-involvements).
+ */
+object Test6124MultipleDefinitionInInvolvements : Test {
+    override fun test(doc: Csaf): ValidationResult {
+        val duplicates =
+            doc.vulnerabilities?.flatMap {
+                (it.involvements?.map { Pair(it.party, it.date) } ?: listOf()).duplicates().keys
+            } ?: listOf()
+
+        return if (duplicates.isEmpty()) {
+            ValidationSuccessful
+        } else {
+            return ValidationFailed(
+                listOf(
+                    "The following party/date pairs are duplicate: ${duplicates.joinToString(", ")}"
+                )
+            )
+        }
+    }
+}
+
+/**
+ * Implementation of
+ * [Test 6.1.25](https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#6125-multiple-use-of-same-hash-algorithm).
+ */
+object Test6125MultipleUseOfSameHashAlgorithm : Test {
+    override fun test(doc: Csaf): ValidationResult {
+        var hashLists = doc.gatherFileHashLists()
+        var duplicates = hashLists.flatMap { it.map { it.algorithm }.duplicates().keys }
+
+        return if (duplicates.isEmpty()) {
+            ValidationSuccessful
+        } else {
+            return ValidationFailed(
+                listOf(
+                    "The following hash algorithms are duplicate: ${duplicates.joinToString(", ")}"
+                )
+            )
+        }
+    }
+}
+
+/**
+ * Implementation of
+ * [Test 6.1.26](https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#6126-prohibited-document-category-name).
+ */
+object Test6126ProhibitedDocumentCategoryName : Test {
+    override fun test(doc: Csaf): ValidationResult {
+        if (doc.document.category in officialProfiles.keys) {
+            return ValidationNotApplicable
+        }
+
+        val cleanedCategory = doc.document.category.lowercase().replace("(_-)", "")
```

Comment:
Should replace all whitespace

Reply:
I am not a regex expert, would `"""(\s_-)""".toRegex()` be the correct term?

Reply:
I'd say `replace("""[\s-_]""".toRegex(), "")` should work

Reply:
Hmm no, test6126 fails now with this :(